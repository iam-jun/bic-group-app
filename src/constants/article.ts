export enum EditArticleErrorType {
  SERIES_DENIED = 'SERIES_DENIED',
  GROUPS_DENIED = 'GROUPS_DENIED',
  TAGS_DENIED = 'TAGS_DENIED',
}

export const EMPTY_ARTICLE_CONTENT = [{ type: 'p', children: [{ text: '' }] }];
