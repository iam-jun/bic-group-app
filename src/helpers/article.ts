import i18next from 'i18next';
import useModalStore from '~/store/modal';
import ImageUploader, { IGetFile, IUploadParam } from '~/services/imageUploader';

export const uploadImage = async ({ file, onSuccess }: {
  file: any, onSuccess: (file: IGetFile) => void
}) => {
  const onError = (error) => {
    const content = typeof error === 'string' ? error : i18next.t('post:error_upload_photo_failed');
    useModalStore.getState().actions.showToast({ content });
  };

  const uploadType = 'post_image';
  const param: IUploadParam = {
    uploadType,
    file,
    onSuccess,
    onError,

  };
  try {
    await ImageUploader.getInstance().upload(param);
  } catch (error) {
    onError(error);
  }
};
