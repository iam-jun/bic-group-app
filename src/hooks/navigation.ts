import { useFocusEffect } from '@react-navigation/native';
import { useCallback, useEffect } from 'react';
import { DeviceEventEmitter } from 'react-native';

import { NAVIGATION_BACK_PRESSED } from '~/configs/navigator';
import { ITabTypes } from '~/interfaces/IRouter';
import { withNavigation } from '~/router/helper';
import { rootNavigationRef } from '~/router/refs';

export const useRootNavigation = () => {
  const rootNavigation = withNavigation(rootNavigationRef);

  return {
    rootNavigation,
  };
};

export const useTabPressListener = (
  callback: (tabName: ITabTypes) => void,
  deps: any[],
) => {
  useEffect(
    () => {
      const listener = DeviceEventEmitter.addListener(
        'onTabPress',
        (tabName: ITabTypes) => {
          callback?.(tabName);
        },
      );
      return () => {
        listener?.remove?.();
      };
    }, deps,
  );
};
/**
 * SHOULD ADD SCREEN TO `customBackHandlerRoutes` FOR USE THIS HOOK
 * @param callback
 */
export const useBackPressListener = (callback: () => void) => {
  useFocusEffect(useCallback(
    () => {
      const backEventListener = DeviceEventEmitter.addListener(
        NAVIGATION_BACK_PRESSED,
        callback,
      );

      return () => {
        backEventListener?.remove?.();
      };
    }, [callback],
  ));
};
