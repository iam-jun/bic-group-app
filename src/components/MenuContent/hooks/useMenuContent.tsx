/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react';
import i18next from 'i18next';
import { isEmpty } from 'lodash';
import Clipboard from '@react-native-clipboard/clipboard';

import homeStack from '~/router/navigator/MainStack/stacks/homeStack/stack';
import { generateLink } from '~/utils/link';
import { IPost, PostType } from '~/interfaces/IPost';
import { IPayloadReactionDetailBottomSheet } from '~/interfaces/IModal';
import { Button } from '~/baseComponents';
import { useRootNavigation } from '../../../hooks/navigation';
import ReportContent from '~/components/Report/ReportContent';
import SeriesContentModal from '~/components/series/SeriesContentModal';
import useCommonController from '~/screens/store';
import { getRootGroupids } from '~/helpers/post';
import { ReportTo } from '~/interfaces/IReport';
import useModalStore from '~/store/modal';
import usePostsStore, { IPostsState } from '~/store/entities/posts';
import { onPressReportThisMember } from '~/helpers/blocking';
import quizStack from '~/router/navigator/MainStack/stacks/quizStack/stack';
import useQuizzesStore from '~/store/entities/quizzes';
import articleStack from '~/router/navigator/MainStack/stacks/articleStack/stack';
import seriesStack from '~/router/navigator/MainStack/stacks/series/stack';
import { getCopyLinkType, getReportContentType } from '../helper';
import useArticleController from '~/screens/articles/store';
import useNotificationItemMenu from '~/screens/Notification/components/NotificationMenu/store';
import { SpecificNotificationType } from '~/interfaces/INotification';

const useMenuContent = (
  data: IPost,
  contentType: PostType,
  isFromDetail: boolean,

  handleConfirmDeleteSeries?: () => void,
  handleDeletePostError?: (listAudiences: string[]) => void,
) => {
  const { rootNavigation } = useRootNavigation();

  const commonActions = useCommonController((state) => state.actions);
  const modalActions = useModalStore((state) => state.actions);
  const actionsQuizzesStore = useQuizzesStore((state) => state.actions);
  const articleControllerActions = useArticleController((state) => state.actions);
  const { deletePost } = usePostsStore((state: IPostsState) => state.actions);
  const specifictNotiActions = useNotificationItemMenu((state) => state.actions);

  const copyLinkType = getCopyLinkType(contentType);
  const reportContentType = getReportContentType(contentType);

  if (!data) return null;

  const {
    id: contentId,
    reactionsCount,
    audience,
    actor,
    quiz,
  } = data;

  const onPressEdit = () => {
    modalActions.hideModal();
    if (contentType === PostType.POST) {
      rootNavigation?.navigate?.(homeStack.createPost, {
        postId: contentId,
        replaceWithDetail: !isFromDetail,
      });
    }

    if (contentType === PostType.ARTICLE) {
      rootNavigation.navigate(articleStack.createArticle, { articleId: contentId });
    }

    if (contentType === PostType.SERIES) {
      rootNavigation?.navigate?.(
        seriesStack.createSeries, {
          seriesId: contentId,
          isFromDetail,
        },
      );
    }
  };

  const onPressEditSettings = () => {
    modalActions.hideModal();
    if (contentType === PostType.POST) {
      rootNavigation?.navigate?.(homeStack.postSettings, {
        postId: contentId,
        isFromPostMenuSettings: true,
      });
    }

    if (contentType === PostType.ARTICLE) {
      rootNavigation?.navigate?.(articleStack.createArticleSettings, {
        articleId: contentId,
        isFromArticleMenuSettings: true,
      });
    }

    if (contentType === PostType.SERIES) {
      rootNavigation?.navigate?.(seriesStack.seriesSettings, {
        seriesId: contentId,
        isFromSeriesMenuSettings: true,
      });
    }
  };

  const onPressSave = (isSaved: boolean) => {
    modalActions.hideModal();
    if (isSaved) {
      commonActions.unsavePost(contentId, contentType);
    } else {
      commonActions.savePost(contentId, contentType);
    }
  };

  const onPressCopyLink = () => {
    modalActions.hideModal();
    Clipboard.setString(generateLink(copyLinkType, contentId));
    modalActions.showToast({ content: 'common:text_link_copied_to_clipboard' });
  };

  const onPressViewReactions = () => {
    modalActions.hideModal();
    const firstReact = reactionsCount[0];
    if (!!firstReact && !isEmpty(firstReact)) {
      const initReaction = Object.keys(firstReact)[0];
      const payload: IPayloadReactionDetailBottomSheet = {
        reactionsCount,
        initReaction,
        getDataParam: { target: 'POST', targetId: contentId },
      };
      modalActions.showReactionDetailBottomSheet(payload);
    }
  };

  const onPressViewSeries = () => {
    modalActions.hideModal();

    modalActions.showModal({
      isOpen: true,
      isFullScreen: true,
      titleFullScreen: i18next.t('common:btn_view_series'),
      ContentComponent: <SeriesContentModal id={contentId} />,
    });
  };

  const onPressPin = () => {
    modalActions.hideModal();
    rootNavigation?.navigate?.(homeStack.pinContent, { postId: contentId });
  };

  const onPressReport = () => {
    const rootGroupIds = getRootGroupids(audience);

    modalActions.hideModal();

    // in this sprint default reportTo is COMMUNITY
    modalActions.showModal({
      isOpen: true,
      ContentComponent: (
        <ReportContent
          targetId={contentId}
          targetType={reportContentType}
          groupIds={rootGroupIds}
          reportTo={ReportTo.COMMUNITY}
        />
      ),
    });
  };

  const _onPressReportThisMember = () => {
    onPressReportThisMember({ modalActions, actor });
  };

  const onPressCUDQuiz = () => {
    modalActions.hideModal();
    rootNavigation?.navigate?.(quizStack.entryQuiz, { postId: contentId });
  };

  const onConfirmEditQuiz = () => {
    const onSuccess = () => {
      rootNavigation?.navigate(
        quizStack.composeQuiz, {
          quizId: quiz?.id,
        },
      );
    };
    actionsQuizzesStore.getQuizDetail({ quizId: quiz?.id, onSuccess });
  };

  const onPressEditQuiz = () => {
    modalActions.hideModal();

    modalActions.showAlert({
      title: i18next.t('quiz:alert_edit:header'),
      content: i18next.t('quiz:alert_edit:content'),
      cancelBtn: true,
      confirmLabel: i18next.t('quiz:continue'),
      onConfirm: onConfirmEditQuiz,
    });
  };

  const onPressDeleteQuiz = () => {
    modalActions.hideModal();

    modalActions.showAlert({
      title: i18next.t('quiz:alert_delete:header'),
      content: i18next.t('quiz:alert_delete:content'),
      cancelBtn: true,
      confirmLabel: i18next.t('common:btn_delete'),
      ConfirmBtnComponent: Button.Danger,
      confirmBtnProps: { type: 'ghost' },
      onConfirm: () => {
        actionsQuizzesStore.deleteQuiz(quiz?.id, contentId);
      },
    });
  };

  const onPressDeleteContent = () => {
    modalActions.hideModal();

    if (contentType === PostType.POST) {
      modalActions.showAlert({
        title: i18next.t('post:title_delete_post'),
        content: i18next.t('post:content_delete_post'),
        cancelBtn: true,
        confirmLabel: i18next.t('common:btn_delete'),
        ConfirmBtnComponent: Button.Danger,
        confirmBtnProps: { type: 'ghost' },
        onConfirm: () => deletePost({
          id: contentId,
          // callbackError: handleDeletePostError,
        }),
      });
    }

    if (contentType === PostType.ARTICLE) {
      modalActions.showAlert({
        title: i18next.t('article:menu:delete'),
        content: i18next.t('post:content_delete_article'),
        cancelBtn: true,
        confirmLabel: i18next.t('common:btn_delete'),
        ConfirmBtnComponent: Button.Danger,
        confirmBtnProps: { type: 'ghost' },
        onConfirm: () => articleControllerActions.deleteArticle(
          contentId,
        ),
      });
    }

    if (contentType === PostType.SERIES) {
      modalActions.showAlert({
        title: i18next.t('series:menu_text_delete_series'),
        content: i18next.t('series:content_delete_series'),
        cancelBtn: true,
        confirmLabel: i18next.t('common:btn_delete'),
        ConfirmBtnComponent: Button.Danger,
        confirmBtnProps: { type: 'ghost' },
        onConfirm: handleConfirmDeleteSeries,
      });
    }
  };

  const onPressNotificationSettingContent = (
    isEnable: boolean,
    contentTargetType: SpecificNotificationType,
  ) => {
    modalActions.hideModal();

    specifictNotiActions.editNotificationSettings(contentId, !isEnable, contentTargetType);
  };

  return {
    onPressEdit,
    onPressEditSettings,
    onPressSave,
    onPressCopyLink,
    onPressViewReactions,
    onPressViewSeries,
    onPressPin,
    onPressReport,
    _onPressReportThisMember,
    onPressCUDQuiz,
    onPressEditQuiz,
    onPressDeleteQuiz,
    onPressDeleteContent,
    onPressNotificationSettingContent,
  };
};

export default useMenuContent;
