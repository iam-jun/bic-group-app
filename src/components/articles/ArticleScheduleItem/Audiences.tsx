import React from 'react';
import { View, StyleSheet, ScrollView } from 'react-native';
import { useTheme } from '@react-navigation/native';
import { useDispatch } from 'react-redux';
import { IPostAudience } from '~/interfaces/IPost';
import { useRootNavigation } from '~/hooks/navigation';
import Text from '~/baseComponents/Text';
import { spacing } from '~/theme';
import ViewSpacing from '~/beinComponents/ViewSpacing';
import { useBaseHook } from '~/hooks';
import mainStack from '~/router/navigator/MainStack/stack';
import Tag from '~/baseComponents/Tag';
import modalActions from '~/storeRedux/modal/actions';
import { PostAudiencesModal } from '~/components/posts';

interface IAudiences {
  audience: IPostAudience;
}

const Audiences: React.FC<IAudiences> = ({ audience }) => {
  const dispatch = useDispatch();
  const theme = useTheme();
  const { colors } = theme;
  const { rootNavigation } = useRootNavigation();
  const { t } = useBaseHook();
  const { groups = [] } = audience || {};

  const navigateToGroup = (groupId: any, communityId: any) => {
    rootNavigation.navigate(mainStack.groupDetail, { groupId, communityId });
  };

  const navigateToCommunity = (communityId: string) => {
    rootNavigation.navigate(mainStack.communityDetail, { communityId });
  };

  const onPressAudience = (item: any) => {
    const { id, communityId, isCommunity } = item || {};
    if (isCommunity && communityId) {
      navigateToCommunity(communityId);
    } else {
      navigateToGroup(id, communityId);
    }
  };

  const onPressMoreItem = () => {
    dispatch(
      modalActions.showModal({
        isOpen: true,
        isFullScreen: true,
        titleFullScreen: t('post:title_post_to'),
        ContentComponent: <PostAudiencesModal data={groups || []} onPressItemAudience={onPressAudience} />,
      }),
    );
  };

  const renderContent = () => {
    if (groups.length === 0) {
      return null;
    }
    return (
      <>
        {renderItem()}
        {renderMoreItem()}
      </>
    );
  };

  const renderItem = () => {
    const firstItem = groups[0];
    return (
      <Tag
        type="neutral"
        size="small"
        label={firstItem.name}
        onActionPress={() => onPressAudience(firstItem)}
        textProps={{ numberOfLines: 1, style: styles.text }}
        testID="audiences.tag_item"
      />
    );
  };

  const renderMoreItem = () => {
    if (groups.length === 1) {
      return null;
    }
    return (
      <Tag
        type="neutral"
        size="small"
        label={renderLabelMoreItem()}
        onActionPress={onPressMoreItem}
        textProps={{ numberOfLines: 1, style: styles.text }}
        testID="audiences.tag_more_item"
      />
    );
  };

  const renderLabelMoreItem = () => t('common:text_other_audience', { count: groups.length - 1 });

  return (
    <View style={styles.boxAudience}>
      <Text.BodyXS color={colors.neutral60}>{`${t('common:audiences')}:`}</Text.BodyXS>
      <ViewSpacing width={spacing.margin.small} />
      <ScrollView horizontal>{renderContent()}</ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  boxAudience: {
    marginTop: spacing.margin.small,
    flexDirection: 'row',
    alignItems: 'center',
  },
  text: {
    maxWidth: 255,
    paddingLeft: spacing.padding.tiny,
  },
});

export default Audiences;
