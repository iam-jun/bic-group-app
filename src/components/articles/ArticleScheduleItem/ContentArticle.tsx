import React from 'react';
import { View, StyleSheet } from 'react-native';
import { useTheme, ExtendedTheme } from '@react-navigation/native';
import { IArticleCover, IAudienceUser, IPostAudience } from '~/interfaces/IPost';
import { useRootNavigation } from '~/hooks/navigation';
import { Avatar, Button } from '~/baseComponents';
import Text from '~/baseComponents/Text';
import Image from '~/components/Image';
import mainTabStack from '~/router/navigator/MainStack/stack';
import { borderRadius } from '~/theme/spacing';
import { spacing } from '~/theme';
import ViewSpacing from '~/beinComponents/ViewSpacing';
import Audiences from './Audiences';

interface ContentArticleProps {
    title: string;
    summary: string;
    coverMedia: IArticleCover;
    actor: IAudienceUser;
    audience: IPostAudience;
    showAvatar: boolean;
}

const ContentArticle: React.FC<ContentArticleProps> = ({
  title,
  summary,
  coverMedia,
  actor,
  audience,
  showAvatar = true,
}) => {
  const theme = useTheme();
  const { colors } = theme;
  const styles = createStyle(theme);
  const { url } = coverMedia || {};
  const { avatar, fullname } = actor || {};
  const { rootNavigation } = useRootNavigation();

  const onPressActor = () => {
    if (!actor?.id) return;
    rootNavigation.navigate(mainTabStack.userProfile, { userId: actor?.id });
  };

  return (
    <View style={styles.container}>
      <Text.H4 color={colors.neutral60} numberOfLines={1}>
        {title}
      </Text.H4>
      <View style={styles.boxInfo}>
        {!!url && <Image style={styles.img} source={url} />}
        {!!summary && (
        <View style={styles.boxSummary}>
          <Text.ParagraphM
            numberOfLines={2}
            color={colors.neutral60}
          >
            {summary}
          </Text.ParagraphM>
        </View>
        )}
      </View>
      {showAvatar && (
        <View style={styles.boxActor}>
          <Button
            style={styles.avatar}
            onPress={onPressActor}
          >
            <Avatar.Tiny isRounded source={avatar} />
            <ViewSpacing width={spacing.margin.small} />
            <View style={styles.boxFullName}>
              <Text.BodyXSMedium color={colors.neutral60} numberOfLines={1}>
                {fullname}
              </Text.BodyXSMedium>
            </View>
          </Button>
        </View>
      )}
      <Audiences audience={audience} />
    </View>
  );
};

const createStyle = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      backgroundColor: colors.white,
      padding: spacing.padding.large,
    },
    boxInfo: {
      flexDirection: 'row',
      alignItems: 'center',
      marginTop: spacing.margin.small,
    },
    img: {
      width: 61,
      height: 44,
      borderRadius: borderRadius.small,
      marginRight: 8,
    },
    boxSummary: {
      flex: 1,
    },
    boxActor: {
      marginTop: spacing.margin.small,
    },
    avatar: {
      flex: 1,
      flexDirection: 'row',
      alignItems: 'center',
    },
    boxFullName: {
      flex: 1,
    },
  });
};

export default ContentArticle;
