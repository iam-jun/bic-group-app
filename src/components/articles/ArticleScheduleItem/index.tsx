import React from 'react';
import { IPost } from '~/interfaces/IPost';
import { Button } from '~/baseComponents';
import ContentArticle from './ContentArticle';
import { ArticleBoxScheduleTime } from '~/components/articles';
import { useRootNavigation } from '~/hooks/navigation';
import articleStack from '~/router/navigator/MainStack/stacks/articleStack/stack';

interface ArticleScheduleItemProps {
    data: IPost;
    showAvatar?: boolean;
    isAdmin?: boolean;
}

const ArticleScheduleItem: React.FC<ArticleScheduleItemProps> = ({ data, showAvatar = true, isAdmin = false }) => {
  const { rootNavigation } = useRootNavigation();

  const {
    title,
    coverMedia,
    summary,
    actor,
    id,
    audience,
    scheduledAt,
    status,
  } = data || {};

  const goToArticleReviewSchedule = () => {
    rootNavigation.navigate(articleStack.articleReviewSchedule, { articleId: id, isAdmin });
  };

  return (
    <Button
      onPress={goToArticleReviewSchedule}
      testID="article_schedule.btn_content"
    >
      <ArticleBoxScheduleTime
        scheduledAt={scheduledAt}
        status={status}
      />
      <ContentArticle
        title={title}
        actor={actor}
        audience={audience}
        coverMedia={coverMedia}
        summary={summary}
        showAvatar={showAvatar}
      />
    </Button>
  );
};

export default ArticleScheduleItem;
