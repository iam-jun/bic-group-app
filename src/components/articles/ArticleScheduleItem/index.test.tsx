import React from 'react';
import { mockArticle } from '~/test/mock_data/article';
import { IPost } from '~/interfaces/IPost';
import { fireEvent, renderWithRedux } from '~/test/testUtils';
import ArticleScheduleItem from '.';
import * as navigationHook from '~/hooks/navigation';
import articleStack from '~/router/navigator/MainStack/stacks/articleStack/stack';
import mainStack from '~/router/navigator/MainStack/stack';

describe('ArticleScheduleItem component', () => {
  it('should navigate to ArticleReviewSchedule screen', () => {
    const navigate = jest.fn();
    const rootNavigation = { navigate };
    jest.spyOn(navigationHook, 'useRootNavigation').mockImplementation(
      () => ({ rootNavigation } as any),
    );

    const rendered = renderWithRedux(
      <ArticleScheduleItem
        data={mockArticle as IPost}
      />,
    );
    const btnContent = rendered.getByTestId('article_schedule.btn_content');
    expect(btnContent).not.toBeNull();
    fireEvent.press(btnContent);
    expect(navigate).toBeCalledWith(articleStack.articleReviewSchedule, { articleId: mockArticle.id, isAdmin: false });

    const tagItem = rendered.getByTestId('audiences.tag_item');
    expect(tagItem).not.toBeNull();
    fireEvent.press(tagItem);
    expect(navigate).toBeCalledWith(mainStack.communityDetail, {
      communityId: mockArticle.audience.groups[0].communityId,
    });

    const tagMoreItem = rendered.getByTestId('audiences.tag_more_item');
    expect(tagMoreItem).not.toBeNull();
    fireEvent.press(tagMoreItem);
  });
});
