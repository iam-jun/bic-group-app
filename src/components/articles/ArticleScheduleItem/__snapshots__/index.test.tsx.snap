// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleScheduleItem component render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="article_schedule.btn_content"
>
  <View
    style={
      Array [
        Object {
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        },
        Object {
          "backgroundColor": "#EBF4FF",
        },
      ]
    }
    testID="article_box_schedule_time"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingRight": 6,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#212745",
              "fontFamily": "BeVietnamPro-Light",
              "fontSize": 11,
            },
            Object {
              "color": "#353E6E",
            },
            undefined,
          ]
        }
      >
        Your article is sheduled to publish:
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 8,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          hitSlop={
            Object {
              "bottom": 10,
              "left": 10,
              "right": 10,
              "top": 10,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "marginRight": 4,
                },
                Object {
                  "backgroundColor": undefined,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              <View />
            </View>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#212745",
                "fontFamily": "BeVietnamPro-Medium",
                "fontSize": 15,
              },
              Object {
                "color": "#007AFF",
              },
              undefined,
            ]
          }
        >
          
        </Text>
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#212745",
              "fontFamily": "BeVietnamPro-Light",
              "fontSize": 14,
            },
            Object {
              "color": "#85C0FF",
            },
            undefined,
          ]
        }
      >
         
        -
         
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          hitSlop={
            Object {
              "bottom": 10,
              "left": 10,
              "right": 10,
              "top": 10,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                },
                Object {
                  "marginRight": 4,
                },
                Object {
                  "backgroundColor": undefined,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              <View />
            </View>
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#212745",
                "fontFamily": "BeVietnamPro-Medium",
                "fontSize": 15,
              },
              Object {
                "color": "#007AFF",
              },
              undefined,
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "padding": 16,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#181C32",
            "fontFamily": "BeVietnamPro-SemiBold",
            "fontSize": 17,
          },
          Object {
            "color": "#282E53",
          },
          undefined,
        ]
      }
    >
      Article thá»© 4 edited 
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 4,
              "height": 44,
              "marginRight": 8,
              "width": 61,
            },
          ]
        }
      >
        <FastImageView
          onFastImageError={[Function]}
          resizeMode="cover"
          source={
            Object {
              "uri": "https://bic-dev-user-upload-images-s3-bucket.s3.ap-southeast-1.amazonaws.com/post/original/fa2a7cde-eee0-4c09-9784-45e1c1ef7e6b.png",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#181C32",
                "fontFamily": "BeVietnamPro-Light",
                "fontSize": 15,
              },
              Object {
                "color": "#282E53",
              },
              undefined,
            ]
          }
        >
          1234567890-= lalalalallalallalalala and I will be there at the
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 8,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "opacity": 1,
          }
        }
        testID="button_wrapper"
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 4,
                "height": 24,
                "width": 24,
              },
              undefined,
            ]
          }
          testID="avatar"
        >
          <View
            style={
              Array [
                Array [
                  Object {
                    "borderRadius": 4,
                    "height": 24,
                    "width": 24,
                  },
                  Object {
                    "borderRadius": 12,
                  },
                ],
                Object {},
              ]
            }
            testID="avatar_container"
          >
            <View
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Array [
                    Array [
                      Object {
                        "borderRadius": 4,
                        "height": 24,
                        "width": 24,
                      },
                      Object {
                        "borderRadius": 12,
                      },
                    ],
                    Object {
                      "borderColor": "#F2F5F8",
                      "borderWidth": 2,
                    },
                    Object {
                      "backgroundColor": "#FFFFFF",
                    },
                  ],
                ]
              }
            >
              <FastImageView
                onFastImageError={[Function]}
                resizeMode="cover"
                source={
                  Object {
                    "uri": "https://bic-dev-entity-attribute-s3-bucket.s3.ap-southeast-1.amazonaws.com/user/avatar/images/original/c6b8b056-7d77-4382-b95c-010c2fda4809.jpg",
                  }
                }
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
                testID="avatar.image"
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": undefined,
              "paddingHorizontal": undefined,
              "paddingVertical": undefined,
              "width": 8,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#212745",
                  "fontFamily": "BeVietnamPro-Medium",
                  "fontSize": 11,
                },
                Object {
                  "color": "#282E53",
                },
                undefined,
              ]
            }
          >
            Nguyen Thi Thu Quyá»n
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginTop": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#212745",
              "fontFamily": "BeVietnamPro-Light",
              "fontSize": 11,
            },
            Object {
              "color": "#282E53",
            },
            undefined,
          ]
        }
      >
        Audiences:
      </Text>
      <View
        style={
          Object {
            "height": undefined,
            "paddingHorizontal": undefined,
            "paddingVertical": undefined,
            "width": 8,
          }
        }
      />
      <RCTScrollView
        horizontal={true}
      >
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F1F2F8",
                "borderRadius": 100,
                "flexDirection": "row",
                "height": 24,
                "justifyContent": "center",
                "marginRight": 8,
                "opacity": 1,
                "paddingLeft": 4,
                "paddingRight": 8,
              }
            }
            testID="audiences.tag_item"
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#212745",
                    "fontFamily": "BeVietnamPro-Light",
                    "fontSize": 11,
                  },
                  Object {},
                  Object {
                    "maxWidth": 255,
                    "paddingLeft": 4,
                  },
                ]
              }
              testID="tag.label"
            >
              Community cá»§a HoÃ ng
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F1F2F8",
                "borderRadius": 100,
                "flexDirection": "row",
                "height": 24,
                "justifyContent": "center",
                "marginRight": 8,
                "opacity": 1,
                "paddingLeft": 4,
                "paddingRight": 8,
              }
            }
            testID="audiences.tag_more_item"
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#212745",
                    "fontFamily": "BeVietnamPro-Light",
                    "fontSize": 11,
                  },
                  Object {},
                  Object {
                    "maxWidth": 255,
                    "paddingLeft": 4,
                  },
                ]
              }
              testID="tag.label"
            >
              + 1 other
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
